File: python/selection_sort.py
"""
Algorithm: Selection Sort
Description: Selects the smallest item iteratively to sort the list; runs in O(n^2) time.
Usage: Run with `python selection_sort.py`.
"""

def selection_sort(data):
    for i in range(len(data)):
        min_index = i
        for j in range(i + 1, len(data)):
            if data[j] < data[min_index]:
                min_index = j
        data[i], data[min_index] = data[min_index], data[i]
    return data


def main():
    values = [7, 3, 9, 2, 5]
    sorted_values = selection_sort(values[:])
    print("Selection sorted values:", sorted_values)


if __name__ == "__main__":
    main()

File: go/selection-sort.go
// Algorithm: Selection Sort
// Description: Repeatedly selects the minimum from the unsorted part; O(n^2) time.
// Usage: Run with `go run selection-sort.go`.
package main

import "fmt"

func selectionSort(data []int) {
    for i := 0; i < len(data); i++ {
        minIndex := i
        for j := i + 1; j < len(data); j++ {
            if data[j] < data[minIndex] {
                minIndex = j
            }
        }
        data[i], data[minIndex] = data[minIndex], data[i]
    }
}

func main() {
    values := []int{5, 3, 6, 2, 1}
    selectionSort(values)
    fmt.Println("Selection sorted values:", values)
}
